specs/sharding/beacon-chain.md

```diff
@@ -103,786 +87,558 @@ The following values are (non-configurable) constants used throughout the specif
-                value=pending_header.attested,
-            )
-```
+        assert intermediate_block_bid.execution_payload_root == hash_tree_root(block.body.execution_payload)
 
-##### `process_shard_header`
+        assert intermediate_block_bid.sharded_data_commitment_number == block.body.sharded_commitments_container.included_sharded_data_commitments
 
-```python
-def process_shard_header(state: BeaconState, signed_header: SignedShardBlobHeader) -> None:
-    header: ShardBlobHeader = signed_header.message
-    slot = header.slot
-    shard = header.shard
-
-    # Verify the header is not 0, and not from the future.
-    assert Slot(0) < slot <= state.slot
-    header_epoch = compute_epoch_at_slot(slot)
-    # Verify that the header is within the processing time window
-    assert header_epoch in [get_previous_epoch(state), get_current_epoch(state)]
-    # Verify that the shard is valid
-    shard_count = get_active_shard_count(state, header_epoch)
-    assert shard < shard_count
-    # Verify that a committee is able to attest this (slot, shard)
-    start_shard = get_start_shard(state, slot)
-    committee_index = (shard_count + shard - start_shard) % shard_count    
-    committees_per_slot = get_committee_count_per_slot(state, header_epoch)
-    assert committee_index <= committees_per_slot
-
-    # Check that this data is still pending
-    committee_work = state.shard_buffer[slot % SHARD_STATE_MEMORY_SLOTS][shard]
-    assert committee_work.status.selector == SHARD_WORK_PENDING
-
-    # Check that this header is not yet in the pending list
-    current_headers: List[PendingShardHeader, MAX_SHARD_HEADERS_PER_SHARD] = committee_work.status.value
-    header_root = hash_tree_root(header)
-    assert header_root not in [pending_header.attested.root for pending_header in current_headers]
-
-    # Verify proposer matches
-    assert header.proposer_index == get_shard_proposer_index(state, slot, shard)
-
-    # Verify builder and proposer aggregate signature
-    blob_signing_root = compute_signing_root(header, get_domain(state, DOMAIN_SHARD_BLOB))
-    builder_pubkey = state.blob_builders[header.builder_index].pubkey
-    proposer_pubkey = state.validators[header.proposer_index].pubkey
-    assert bls.FastAggregateVerify([builder_pubkey, proposer_pubkey], blob_signing_root, signed_header.signature)
-
-    # Verify the length by verifying the degree.
-    body_summary = header.body_summary
-    points_count = body_summary.commitment.samples_count * POINTS_PER_SAMPLE
-    if points_count == 0:
-        assert body_summary.degree_proof == G1_SETUP[0]
-    assert (
-        bls.Pairing(body_summary.degree_proof, G2_SETUP[0])
-        == bls.Pairing(body_summary.commitment.point, G2_SETUP[-points_count])
-    )
+        assert intermediate_block_bid.sharded_data_commitment_root == hash_tree_root(block.body.sharded_commitments_container.sharded_commitments[-intermediate_block_bid.sharded_data_commitments:])
```

Comment:
I think the index access of `sharded_commitments` might not be right. Should that be `included_sharded_data_commitments`?